week2/materials/homework/webapp/src/hooks/useProducts.js

```diff
@@ -37,17 +37,30 @@ let initialProducts = productsData.map((item) => {
 function useProducts() {
   const [products] = useState(initialProducts);
   const [cart, setCart] = useState([]);
+  // const [priceSum, setPriceSum] = useState()
 
   const addProduct = (product) => {
     let newCart = cart.concat(product);
     setCart(newCart);
   };
 
+  const calculateSum = ()=> { 
+    // let totalPriceOfProduct= 0;
+    // let totalProducts= 
+    return cart.reduce((carry, product)=>{
+      return carry + parseInt(product.price)
+    },0);
+
+    //   parseInt(product.price), totalPriceOfProduct);
+    // setCart(totalProducts);
+    
+  }
+
   const removeProduct = (product) => {
     setCart(cart.filter((item) => item.id != product.id));
   };
 
-  return { products, cart, addProduct, removeProduct };
+  return { products, cart, addProduct, removeProduct, Total: calculateSum() };
```

Comment:
```suggestion
  return { products, cart, addProduct, removeProduct, total: calculateSum() };
```

Reply:
Thank you soo much @bytewiz  for your time on reviewing my homework. :)